### ISSUE TITLE
'loss is NaN' problem

### ISSUE THREAD

**Issue by mengxingkong** [jingxil/Neural-Decision-Forests#8]
while updating pi with theta fixed, the result of new_pi in the first step is too large,  using softmax to normalize  new_pi will get output near to zero. And in next step wo get NAN new_pi

**Comment by mengxingkong** [comment]
#new_pi = F.softmax(Variable(new_pi), dim=1).data
new_pi = F.normalize(new_pi, p=1,dim=1)

**Comment by emadeldeen24** [comment]
check the activation function ... here it uses Sigmoid .. according to your problem you may need to change it maybe to ReLU
